File: androidbase-sdk/src/main/java/com/leovp/androidbase/annotations/Density.java
package com.leovp.androidbase.annotations;

import androidx.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@IntDef({ Density.WIDTH_BASED, Density.HEIGHT_BASED, Density.LONG_SIDE_BASED, Density.SHORT_SIDE_BASED })
@Retention(RetentionPolicy.SOURCE)
public @interface Density {
    int WIDTH_BASED = 1;
    int HEIGHT_BASED = 2;
    int LONG_SIDE_BASED = 3;
    int SHORT_SIDE_BASED = 4;
}

File: androidbase-sdk/src/main/java/com/leovp/androidbase/annotations/Density.kt
package com.leovp.androidbase.annotations

import androidx.annotation.IntDef

/**
 * Author: Michael Leo
 * Date: 21-1-26 下午3:19
 */
@IntDef(Density.WIDTH_BASED, Density.HEIGHT_BASED, Density.LONG_SIDE_BASED, Density.SHORT_SIDE_BASED)
@kotlin.annotation.Retention(AnnotationRetention.SOURCE)
annotation class Density {
    companion object {
        const val WIDTH_BASED = 1
        const val HEIGHT_BASED = 2
        const val LONG_SIDE_BASED = 3
        const val SHORT_SIDE_BASED = 4
    }
}
